tests/codegen/intcast_assign_var.cpp

```diff
@@ -0,0 +1,48 @@
+#include "common.h"
+
+namespace bpftrace {
+namespace test {
+namespace codegen {
+
+TEST(codegen, intcast_retval)
+{
+  // Make sure the result is truncated to 32 bit and sign extended to 64
+  test("kretprobe:f { @=(int32)retval }",
+R"EXPECTED(; Function Attrs: nounwind
+declare i64 @llvm.bpf.pseudo(i64, i64) #0
+
+; Function Attrs: argmemonly nounwind
+declare void @llvm.lifetime.start.p0i8(i64, i8* nocapture) #1
+
+define i64 @"kretprobe:f"(i8* nocapture readonly) local_unnamed_addr section "s_kretprobe:f_1" {
+entry:
+  %"@_val" = alloca i64, align 8
+  %"@_key" = alloca i64, align 8
+  %1 = getelementptr i8, i8* %0, i64 80
+  %retval = load i64, i8* %1, align 8
+  %2 = bitcast i64* %"@_key" to i8*
+  call void @llvm.lifetime.start.p0i8(i64 -1, i8* nonnull %2)
+  store i64 0, i64* %"@_key", align 8
+  %sext = shl i64 %retval, 32
+  %3 = ashr exact i64 %sext, 32
```

Comment:
Assuming these are the instructions we care about.

Side note: would be nice if we had some regex support so we could `*` out the other less important bits.